### ISSUE TITLE
Npm Start Error

### ISSUE THREAD

**Issue by dehlers1000** [chrisjwalk/angular-cli-netcore-ngrx-starter#8]
Hi Chris,

Awesome work on organizing a dotnet application in a NX workspace. This is the first time I have seen this done. 

When doing an npm start, I get the following error:

> ng run api:serve 

> ng run web-app:serve 
Executing "dotnet run --project apps/api"...
crit
: Microsoft.AspNetCore.Hosting.Diagnostics[6]
      Application startup exception
System.IO.DirectoryNotFoundException: C:\Architect\Nx\dotnetnx\apps\api\apps\web-app\
   at Microsoft.Extensions.FileProviders.PhysicalFileProvider..ctor(String root, ExclusionFilters filters)
   at Microsoft.Extensions.FileProviders.PhysicalFileProvider..ctor(String root)
   at Api.Startup.Configure(IApplicationBuilder app, IWebHostEnvironment env) in C:\Architect\Nx\dotnetnx\apps\api\Startup.cs:line 51
   at System.RuntimeMethodHandle.InvokeMethod(Object target, Object[] arguments, Signature sig, Boolean constructor, Boolean wrapExceptions)     
   at System.Reflection.RuntimeMethodInfo.Invoke(Object obj, BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)   
   at Microsoft.AspNetCore.Hosting.ConfigureBuilder.Invoke(Object instance, IApplicationBuilder builder)
   at Microsoft.AspNetCore.Hosting.ConfigureBuilder.<>c__DisplayClass4_0.<Build>b__0(IApplicationBuilder builder)
   at Microsoft.AspNetCore.Hosting.GenericWebHostBuilder.<>c__DisplayClass13_0.<UseStartup>b__2(IApplicationBuilder app)
   at Microsoft.AspNetCore.Mvc.Filters.MiddlewareFilterBuilderStartupFilter.<>c__DisplayClass0_0.<Configure>g__MiddlewareFilterBuilder|0(IApplicationBuilder builder)
   at Microsoft.AspNetCore.HostFilteringStartupFilter.<>c__DisplayClass0_0.<Configure>b__0(IApplicationBuilder app)
   at Microsoft.AspNetCore.Hosting.GenericWebHostService.StartAsync(CancellationToken cancellationToken)

Unhandled exception. 
An unexpected error occurred: undefined

Do recognize this error? Maybe I am doing something wroing.

Thanks,
Danny

**Comment by chrisjwalk** [comment]
Hi Danny, looks like you just found a bug! 

TLDR, we just need to create an empty folder at `.\apps\api\apps\web-app` for the `npm start` to work. If you pull the latest, this should be fixed

If you look at the files in the `./dist` folder after doing a `npm build`, you'd see that the "web-app" dist files are in `./dist/apps/web-app/` folder, and the "api" dist files are in the root of the `./dist` folder. This is because when published, the dotnet "api" project is also serving the "web-app" dist as static files using Microsoft.AspNetCore.SpaServices. This allows the files in that folder (including the default "index.html" for all other unknown routes), and the routes from the API Controllers to all be served off of the root when deployed.

However, when running this in development mode, the angular cli dev server is instead the one acting as the main server and API routes are being served through a proxy off of the /api folder as defined in the "proxy.conf.json" file :

https://github.com/chrisjwalk/angular-cli-netcore-ngrx-starter/blob/master/apps/web-app/proxy.conf.json

When running the dotnet "api" project in dev mode, we don't care about the static files from the `./apps/web-app` folder. If that folder doesn't exist however, you'll get the exception you were seeing. So the (mostly) empty folder added here should fix this:

https://github.com/chrisjwalk/angular-cli-netcore-ngrx-starter/commit/95953c69850421b6c7e5befdba4bb26d9d2a210c

**Comment by dehlers1000** [comment]
Thanks Chris. I pulled the latest code and the problem is resolved. I appreciate the explanation on how it works.
